src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTests.cs

```diff
@@ -0,0 +1,317 @@
+                    "--type", "toolset",
+                    "--repo", sourceRepoUri);
+                await RunGitAsync("commit", "-am", "Add dependencies.");
+                await using IAsyncDisposable ___ = await PushGitBranchAsync("origin", targetBranch);
+                await TriggerSubscriptionAsync(sub.Value);
+
+                PullRequest pr = await WaitForPullRequestAsync(targetRepo, targetBranch);
+
+                Assert.Equal($"[{targetBranch}] Update dependencies from dotnet/arcade", pr.Title);
+
+                await CheckoutRemoteRefAsync(pr.MergeCommitSha);
+
+                string dependencies = await RunDarcAsync("get-dependencies");
+                string[] dependencyLines = dependencies.Split(new[] {'\n', '\r'}, StringSplitOptions.RemoveEmptyEntries);
+                Assert.Equal(new[]
+                {
+                    "Name:             Microsoft.DotNet.Arcade.Sdk",
+                    "Version:          2.1.0",
+                    $"Repo:             {sourceRepoUri}",
+                    $"Commit:           {sourceCommit}",
+                    "Type:             Toolset",
+                    "Pinned:           False",
+                }, dependencyLines);
+
+                using TemporaryDirectory arcadeRepo = await CloneRepositoryAsync("dotnet", sourceRepo);
+                using (ChangeDirectory(arcadeRepo.Directory))
+                {
+                    await CheckoutRemoteRefAsync(sourceCommit);
+                }
+
+                var arcadeFiles = Directory.EnumerateFileSystemEntries(Path.Join(arcadeRepo.Directory, "eng", "common"),
+                        "*", SearchOption.AllDirectories)
+                    .Select(s => s.Substring(arcadeRepo.Directory.Length))
+                    .ToHashSet();
+                var repoFiles = Directory.EnumerateFileSystemEntries(Path.Join(repo.Directory, "eng", "common"), "*",
+                        SearchOption.AllDirectories)
+                    .Select(s => s.Substring(repo.Directory.Length))
+                    .ToHashSet();
+
+                Assert.Empty(arcadeFiles.Except(repoFiles));
+                Assert.Empty(repoFiles.Except(arcadeFiles));
+            }
+        }
+
+        private async Task<PullRequest> WaitForPullRequestAsync(string targetRepo, string targetBranch)
+        {
+            Repository repo = await GitHubApi.Repository.Get(_parameters.GitHubTestOrg, targetRepo).ConfigureAwait(false);
+            var attempts = 10;
+            while (attempts-- > 0)
+            {
+                IReadOnlyList<PullRequest> prs = await GitHubApi.PullRequest.GetAllForRepository(repo.Id, new PullRequestRequest
+                {
+                    State = ItemStateFilter.Open,
+                    Base = targetBranch,
+                }).ConfigureAwait(false);
+                if (prs.Count == 1)
+                {
+                    return prs[0];
+                }
+                if (prs.Count > 1)
```

Comment:
Xunit is going to run these in parallel right? Most of the previous set of tests were basically constructed as easily as possible, based on the idea that only one test would run at once.  I think this assumption will be invalid in a parallel xunit  world, or a world where we run more than one set of scenario tests at once.

Reply:
Yea it will, but I think thats still fine. The target branch for the PR is a random number so each test will have a different target branch.

Reply:
Right, doesn't this mean that if two PRs from two tests open at once, even though they end up against different branches, both tests will fail.

Reply:
Yea, the target branch filtering should make that not an issue.